lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -126,15 +130,20 @@ def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
         try:
             index_map = Metadata.getvalue(dataset=dataset, key=Metadata.INDEX_MAP)
         except MetadataError as e:
-            self.logger.error(f"Indices from metadata table not found {e!r}")
-            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
-
-        if index_map is None:
             self.logger.error(
                 f"server.index-map not found in Metadata for a dataset {dataset!r}"
             )
-            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+            raise APIAbort(
```

Comment:
And here we're doing a `logger.error` instead of `logger.exception` so we don't have the original cause. (Which could be a SQL error or something "interesting".)